real_robots/omnirobot_utils/marker_finder.py

```diff
@@ -0,0 +1,386 @@
+                        k0 = k
+                    if pt2[0][0] == pt1[0][0] and pt2[0][1] == pt1[0][1]:
+                        k1 = k
+
+                # compute how much points are in this line
+                if k1 >= k0:
+                    length = k1 - k0 - 1
+                else:
+                    length = len(candidate_contours[i]) - k0 + k1 - 1
+
+                if length == 0:
+                    length = 1
+
+                line_pts = np.zeros((1, length, 2), np.float32)
+                # append this line's point to array 'line_pts'
+                for l in range(length):
+                    ll = (k0 + l + 1) % len(candidate_contours[i])
+                    line_pts[0, l, :] = candidate_contours[i][ll]
+
+                # Fit edge and put to vector of edges
+                [vx, vy, x, y] = cv2.fitLine(line_pts, cv2.DIST_L2, 0, 0.01, 0.01)
+                fitted_lines[j, :] = [vx, vy, x, y]
+                if visualise:
+                    # Finally draw the line
+                    # Now find two extreme points on the line to draw line
+                    left = [0, 0]
+                    right = [0, 0]
+                    length = 100
+                    left[0] = x - vx * length
+                    left[1] = y - vy * length
+                    right[0] = x + vx * length
+                    right[1] = y + vy * length
+                    cv2.line(img, tuple(left), tuple(right), 255, 2)
+
+            # Calculated four intersection points
+            for j in range(4):
+                intc = self.intersection(fitted_lines[j, :], fitted_lines[(j+1) % 4, :])
+                self.blob_corners[i, j, :] = intc
+
+            if visualise:
+                for j in range(4):
+                    intc = tuple(self.blob_corners[i, j, :])
+                    if j == 0:
+                        cv2.circle(img, intc, 5, (255, 255, 255))
+                    if j == 1:
+                        cv2.circle(img, intc, 5, (255, 0, 0))
+                    if j == 2:
+                        cv2.circle(img, intc, 5, (0, 255, 0))
+                    if j == 3:
+                        cv2.circle(img, intc, 5, (0, 0, 255))
+                cv2.imshow('frame', img)
+                cv2.waitKey(0)
+                cv2.destroyAllWindows()
+
+    def decode(self, transformed_img):
+        """
+        TODO
+        :param transformed_img:
+        :return:
+        """
+        num_step = 9
+        code = np.zeros((num_step, num_step), np.uint8)
+        step = int(transformed_img.shape[0] / num_step)
+        for i in range(num_step):
+            for j in range(num_step):
+                if np.average(transformed_img[i * step: i * step + step, j * step: j * step + step]) < 100:
+                    code[i, j] = 1
+                else:
+                    code[i, j] = 0
+        return code
+                
+    def rotateCorners(self, corner, i):
+        """
+        TODO
```

Comment:
missing docstring

Reply:
Same for a bunch of method in this file...

Reply:
Hi @GaspardQin, can you please help comment this file ? (see TODO in 
real_robots/omnirobot_utils/marker_finder.py), as I lack context for some of the functions. Thanks